sources/rest_api/__init__.py

```diff
@@ -254,21 +257,39 @@ def paginate_resource(
                 client: RESTClient = client,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
                 incremental_param: IncrementalParam = incremental_param,
+                row_filter: Any = row_filter,
+                transform: Any = transform,
+                exclude_columns: List[str] = exclude_columns,
             ) -> Generator[Any, None, None]:
+                def exclude_elements(item: Any, exclude_columns: List[str]) -> Any:
+                    for key in exclude_columns:
+                        del item[key]
+                    return item
+
                 if incremental_object:
                     params[incremental_param.start] = incremental_object.last_value
                     if incremental_param.end:
                         params[incremental_param.end] = incremental_object.end_value
 
-                yield from client.paginate(
+                for page in client.paginate(
                     method=method,
                     path=path,
                     params=params,
-                    json=json,
                     paginator=paginator,
                     data_selector=data_selector,
                     hooks=hooks,
-                )
+                ):
+                    filtered_page = filter(row_filter, page) if row_filter else page
```

Comment:
```python
yield map(exclude_elements, map(transform, filter(row_filter, page)), exclude_columns)
```

Given an identity function which removes the need for ternaries, I think this is more aesthetic :) 
Closer to an applicative functor on the list monad, though itd look more accurate if python supported map/filter as methods on the list class. Then itd be left to right instead of bubbling out.

Reply:
not in one line, but I did something similar